### ISSUE TITLE
Status code for 403 ApiResponseExcepton

### ISSUE THREAD

**Issue by zaporylie** [SparkPost/php-sparkpost#83]
I saw #71 has been merge into master and that's great! Thanks @j4m3s for your work! 

I'm wondering why it doesn't include status code in APIResponseException? Is there any reason why status code has been omitted?

Would be nice to have it back for 403 responses. Actually it would be even better to add missing data from https://support.sparkpost.com/customer/en/portal/articles/2140916-extended-error-codes like api error code, message and description.

Change https://github.com/SparkPost/php-sparkpost/blob/master/lib/SparkPost/APIResource.php#L197 to:
```
        $response = json_decode($response->getBody(), true);
        throw new APIResponseException(
          'Request forbidden',
          $statusCode,
          isset($response['errors'][0]['message']) ? $response['errors'][0]['message'] : "Request forbidden",
          isset($response['errors'][0]['code']) ? $response['errors'][0]['code'] : 1100,
          isset($response['errors'][0]['description']) ? $response['errors'][0]['description'] : "Does this API Key have the correct SparkPost permissions?"
        );
```

**Comment by beardyman** [comment]
The api http status code is in `$exception->code`, the `$exception->apiCode` is an extended code which may or may not be supplied.  Description is also optional in our responses.

**Comment by zaporylie** [comment]
Yeah, but right now neither `$exception->code` nor `$exception->apiCode` is available for `$statusCode == 403`. See https://github.com/SparkPost/php-sparkpost/blob/master/lib/SparkPost/APIResource.php#L197

**Comment by beardyman** [comment]
Ah I see, sorry for the misunderstanding.  I will address for the next release.

**Comment by richleland** [comment]
This has been published with 1.2.0.
